### PR TITLE
fix: disallow unknown keys in deno.json

### PR DESCRIPTION
**Pull request by dsherret**
I believe this was an oversight when implementing deno.json as we did put this tag on the other structs.

The reasoning for disallowing unknown keys is mainily to prevent users from adding custom keys that could potentially confict with a runtime feature, but also it reduces confusion when a user accidentally mispells a key as they get an error message.

Draft because I'm not sure we're going to do this or maybe we should warn instead?

### CODE REVIEW

File: cli/tests/integration/vendor_tests.rs [hunk 1]
```diff
@@ -160,10 +160,6 @@ fn import_map_output_dir() {
     // will be ignored
     "{ \"imports\": { \"https://localhost:4545/\": \"./localhost/\" }}",
   );
-  t.write(
-    "deno.json",
-    "{ \"import_map\": \"./vendor/import_map.json\" }",
```
Comment (dsherret) [inline cli/tests/integration/vendor_tests.rs:165]: Incorrect key this caught. This is specified via a CLI arg below.

**Comment by kitsonk** [comment]
Disallowing it seems harsh.  _maybe_ warning, but again, why be cruel to people?  It was originally simply reading the key of `compilerOptions` from `--config` for tsconfig.json, where we were already ignoring excess keys, and `deno.json`/`deno.jsonc` was implemented in a way that people could continue to use/maintain a `tsconfig.json` and have it work with Deno.  Restricting or warning would effectively break/change this behaviour.  Then if you allow all the `tsconfig.json` keys without breaking or warning... 😕 

So I am quite 👎  for this.

**Comment by dsherret** [comment]
I forgot about the initial taconfig.json implementation. Given that and the way this has worked, I agree that an error is too harsh. Given that people might use this with an existing tsconfig, yeah maybe even a warning wouldn’t be ideal. The issue with opening up configs to piggy backing is you leave a lot of trust up to other tools to be respectful with that piggy backing. It only takes one other tool to piggy back in a way that makes the owner of the config (the cli) no longer free to use certain config keys because of the ecosystem breakage that would happen. The chance of that happening is low, but it’s there.

If we were designing this from the ground up I’d argue for this, but given this is not actually an oversight, I’m going to close this PR and just hope other tools are respectful.

**Comment by arnauorriols** [comment]
FTR, [denoland/deployctl](https://github.com/denoland/deployctl) now depends on unknown keys being allowed.

**Pull request closed without merging**